api/src/operations/mail/index.ts

```diff
@@ -0,0 +1,26 @@
+import { defineOperationApi } from '@directus/shared/utils';
+import { MailService } from '../../services';
+
+type Options = {
+	template: string;
+	data: string;
+	to: string;
+	subject: string;
+};
+
+export default defineOperationApi<Options>({
+	id: 'mail',
+
+	handler: async ({ template, data, to, subject }, { accountability, database, getSchema }) => {
+		const mailService = new MailService({ schema: await getSchema({ database }), accountability, knex: database });
+
+		await mailService.send({
+			template: {
+				name: template,
+				data: JSON.parse(data),
+			},
+			to,
```

Comment:
Can't get emails to fire. SO far, I have tried:

hardcoding: `eron@directus.io`
Dynamic vars: `$CURRENT_USER.email`
Nils' Syntax as shown in the PR image: `sale.user.email`

**However, if Nils' syntax is the only one permissable, I'm Not sure where the scoping begins. He first uses the plural sales.item.create then later uses sale.user.email and i'm not sure if the singular `sale` is a separate collection, a field, an item, etc.

Also I noticed that in this code, the function takes in a string, in nils' pic the emails appear in an object literal, and in the app they appear in an array.

<img width="926" alt="Screen Shot 2022-05-09 at 11 43 25 AM" src="https://user-images.githubusercontent.com/62306630/167457590-073b5e3c-e13f-418f-9301-076238e8b660.png">
  
pls 🙏  lmk if this is a user-error or bug 🔢

Reply:
Found the problem 👍🏻 There were indeed a couple bugs in the mail implementation

Reply:
Resolved in a9bf178f7